lib/src/cloud_firestore_mocks_base.dart

```diff
@@ -24,6 +25,14 @@ class MockFirestoreInstance extends Mock implements FirebaseFirestore {
     _setupFieldValueFactory();
   }
 
+  /// Clears the state
+  @visibleForTesting
+  void reset() {
+    _root.clear();
```

Comment:
In your tests, instead of clearing the state, would it be possible for you to re-instantiate a MockFirestoreInstance? It would be much safer than trying to recycle one instance. Recycling a MockFirestoreInstance is difficult. For example your implementation doesn't reset _savedDocumentPaths properly. Also the library does a bunch of things with listeners.

I'd prefer not to add a `reset()` method if possible.